### PR TITLE
fix(next-auth): allow react@19 to peer deps

### PR DESCRIPTION
**Pull request by devjiwonchoi**
## ☕️ Reasoning

`next@canary` peerDeps have React19, so add them.

```
└─┬ next-auth 4.24.7
  ├── ✕ unmet peer react@"^17.0.2 || ^18": found 19.0.0-beta-4508873393-20240430
  └── ✕ unmet peer react-dom@"^17.0.2 || ^18": found 19.0.0-beta-4508873393-20240430
```

## 🧢 Checklist

- [ ] Documentation
- [ ] Tests
- [ ] Ready to be merged

## 📌 Resources

- [Security guidelines](https://github.com/nextauthjs/.github/blob/main/SECURITY.md)
- [Contributing guidelines](https://github.com/nextauthjs/.github/blob/main/CONTRIBUTING.md)
- [Code of conduct](https://github.com/nextauthjs/.github/blob/main/CODE_OF_CONDUCT.md)
- [Contributing to Open Source](https://kcd.im/pull-request)

### CODE REVIEW

File: packages/next-auth/package.json [hunk 1]
```diff
@@ -84,7 +84,7 @@
     "@simplewebauthn/server": "^9.0.2",
     "next": "^14",
     "nodemailer": "^6.6.5",
-    "react": "^18.2.0"
+    "react": "^18.2.0 || ^19.0.0-0"
```
Comment (ndom91) [inline packages/next-auth/package.json:87]: This should cover more `19.x` things for now and in the future

```suggestion
    "react": "^18.2.0 || ^19"
```

Comment (devjiwonchoi) [reply packages/next-auth/package.json:87]: AFAIK caret-only `^` doesn't cover the pre-release as well, it had to be set explicitly with `-`.

x-ref: https://semver.npmjs.com

Comment (ndom91) [reply packages/next-auth/package.json:87]: Ah okay, makes sense. Thanks for the npm semver calc link

**Review by ndom91 — APPROVED** [review]

File: packages/next-auth/package.json (hunk 1 above)
Comment (devjiwonchoi) [reply packages/next-auth/package.json:87]: Thank you for reviewing!

**Pull request merged**